### DISCUSSION TITLE
An activity bar icon to view all your scripts in one place

### DISCUSSION THREAD

**Discussion by nixin72** [BetterThanTomorrow/joyride#30] [Ideas]
In VSCode, it can be kind of a pain sometimes using multi-folder workspace because it can break some tools. This makes working with user scripts kind of annoying, because you'd need to switch workspaces to contribute to them. What I'd like to propose is a joyride button in the activity bar that, when clicked, will show you all of your joyride scripts separated into two categories:

- your workspace scripts for the currently active workspace (assuming there are any)
- your user scripts that are in `config/user-scripts-path` 

This would allow you to have very quick and easy access to all your scripts rather than needing to navigate to the specific file you wanted to open, or changing workspaces completely. 

Relevant vscode docs for adding content to the activity bar: https://code.visualstudio.com/api/extension-guides/tree-view

**Comment by PEZ** [comment]
Hello! Thanks for this discussion starter.

Some questions:

* Do you want to open the scripts for editing or do you want to run them?
* Workspace scripts. Do you want to access scripts from all workspace folders, or just the ”active” one?

**Reply by nixin72** [reply]
- I was imagining that you could open and edit the script from the sidebar - being able to quickly run them from there too would also be nice.
- I'd been imagining just the active workspace. I don't see much benefit to editing a workspace-specific script from a different workspace? Although if there's a way of obtaining all known workspaces, it could be a setting? 🤷 

Since workspace scripts are obviously easily available in the explorer too, I'd say an MVP for this would just be listing the user scripts and allowing people to open and edit them. Then running scripts, then adding workspace scripts I guess?
